pkg/controllers/cloud_config_sync_controller_test.go

```diff
@@ -551,3 +560,119 @@ var _ = Describe("Cloud config sync reconciler", func() {
 		Expect(err.Error()).Should(BeEquivalentTo("platformStatus is required"))
 	})
 })
+
+var _ = Describe("Feature gate handling in CloudConfigReconciler", func() {
+	var reconciler *CloudConfigReconciler
+	ctx := context.Background()
+
+	BeforeEach(func() {
+		reconciler = &CloudConfigReconciler{
+			ClusterOperatorStatusClient: ClusterOperatorStatusClient{
+				Client:           cl,
+				Clock:            clocktesting.NewFakePassiveClock(time.Now()),
+				ManagedNamespace: testManagedNamespace,
+			},
+			Scheme:            scheme.Scheme,
+			FeatureGateAccess: featuregates.NewHardcodedFeatureGateAccessForTesting(nil, nil, nil, nil),
+		}
+
+		// Create required infrastructure resource
+		infraResource := makeInfrastructureResource(configv1.AzurePlatformType)
+		Expect(cl.Create(ctx, infraResource)).To(Succeed())
+		infraResource.Status = makeInfraStatus(infraResource.Spec.PlatformSpec.Type)
+		Expect(cl.Status().Update(ctx, infraResource.DeepCopy())).To(Succeed())
+
+		// Create required network resource
+		networkResource := makeNetworkResource()
+		Expect(cl.Create(ctx, networkResource)).To(Succeed())
+
+		// Create source config map
+		Expect(cl.Create(ctx, makeInfraCloudConfig(configv1.AzurePlatformType))).To(Succeed())
+	})
+
+	AfterEach(func() {
+		// Clean up test resources
+		deleteOptions := &client.DeleteOptions{
+			GracePeriodSeconds: ptr.To[int64](0),
+		}
+
+		// Clean up infrastructure
+		infra := &configv1.Infrastructure{
+			ObjectMeta: metav1.ObjectMeta{Name: infrastructureResourceName},
+		}
+		cl.Delete(ctx, infra, deleteOptions) //nolint:errcheck
```

Comment:
I would still check that these succeed. If they fail, they may affect later test cases no?

Is this cleanup similar to how we are handling it in other tests in this repo?